synapse/handlers/federation.py

```diff
@@ -1708,7 +1708,7 @@ async def _sync_partial_state_room(
                             destination, event
                         )
                         break
-                    except FederationError as e:
+                    except (NotRetryingDestination, FederationError) as e:
```

Comment:
I have no idea if there are other exceptions we ought to catch here.

Reply:
I think this is redundant.

`update_state_for_partial_state_event` calls `_compute_event_context_with_maybe_missing_prevs`, which catches all `Exceptions` and turns them into `FederationError`s. 

I don't think there's anywhere else in `update_state_for_partial_state_event` that can raise a `NotRetryingDestination`.

Reply:
In that case I'm confused about what you meant by

> So, I'd just ignore destinations for now, and as @MadLittleMods, just make sure we catch the relevant exceptions, and move on.

in https://github.com/matrix-org/synapse/issues/12999#issuecomment-1273325438. Perhaps the point is that your reasoning in the comment above proves that we're catching the relevant exceptions?

In any case, 29b362dad reverts this.

Reply:
Yes, it looks like we're *already* catching the relevant exceptions. I was just saying we should check that.